src/test/base58_tests.cpp

```diff
@@ -56,15 +57,20 @@ BOOST_AUTO_TEST_CASE(base58_DecodeBase58)
         }
         std::vector<unsigned char> expected = ParseHex(test[0].get_str());
         std::string base58string = test[1].get_str();
-        BOOST_CHECK_MESSAGE(DecodeBase58(base58string, result, 256), strTest);
-        BOOST_CHECK_MESSAGE(result.size() == expected.size() && std::equal(result.begin(), result.end(), expected.begin()), strTest);
+        BOOST_TEST_MESSAGE("Test input: `" << base58string << "`");
+
+        BOOST_CHECK(DecodeBase58(base58string, result, 256));
+        BOOST_CHECK_MESSAGE(
+            result == expected,
+            strTest << "\nDecoding `" << base58string << "` as `" << HexStr(result) << "` should match `" << HexStr(expected) << "`"
```

Comment:
```suggestion
            strTest << ": got \"" << EncodeBase58(sourcedata) << "\""
```
Same suggestion as above.

Reply:
```
check '["271F35A1","211112"]: got "271f35a1"' has passed
```